### PR TITLE
Swagger - update API Documentation: capping support

### PR DESCRIPTION
**Pull request by padv2010**
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDLP-3108

### Description

Documents API changes after adding Capping support.

### CODE REVIEW

File: docs/source/release-notes.html.md [hunk 1]
```diff
@@ -7,6 +7,24 @@ weight: 8
 
 If you have any questions or comments about these notes, please contact DfE via Slack or email.
 
+## 03 Jun 2024
+
+We're trialing new functionality in the API test ([sandbox](https://sb.manage-training-for-early-career-teachers.education.gov.uk/)) environment which will enable providers to set whether NPQ applicants are going to have their training funded by DfE.
+
+This is because from the 2024/25 academic year onwards there'll be a set maximum number of places each provider can offer per NPQ that DfE will pay for.
+
+Providers using all versions of the API can set the new funded_place field in the 'Accept an application' request body to true or false. They will also see the funded_place field in the following endpoint response bodies:
```
Comment (cwrw) [inline docs/source/release-notes.html.md:16]:
```suggestion
Providers using all versions of the API can set the new `funded_place` field in the 'Accept an application' request body to true or false. They will also see the `funded_place` field in the following endpoint response bodies:
```

File: swagger/v3/api_spec.json [hunk 2]
```diff
@@ -161,12 +161,12 @@
         }
       }
     },
-    "/api/v3/partnerships/ecf": {
```
Comment (cwrw) [inline swagger/v3/api_spec.json:164]: could we avoid running swaggerise on v3, and amending manually, as this is the only version that has not been fixed properly, and will change up the whole file?

**Review by leandroalemao — CHANGES_REQUESTED** [review]
Hi @padv2010 there're new fields added on v1/v2/v3 `npq_participant_serializer.rb` so some extra changes in the related docs are needed as well ie: https://github.com/DFE-Digital/early-careers-framework/blob/main/app/serializers/api/v1/npq_participant_serializer.rb#L26

**Comment by padv2010** [comment]
Thanks for the review @cwrw @leandroalemao 

@cwrw I have addressed both changes, 

@leandroalemao, I have updated v1 for ParticipantProfile, V2 and V3, unless I am missing something, are already there. I would appreciate some help though, because I can't add a new Schema object to V1 to support `funded_places`

File: docs/source/release-notes.html.md [hunk 3]
```diff
@@ -7,6 +7,25 @@ weight: 8
 
 If you have any questions or comments about these notes, please contact DfE via Slack or email.
 
+## ?? (pending sandbox release)
```
Comment (padv2010) [inline docs/source/release-notes.html.md:10]:
```suggestion
## 14 June 2024 (pending sandbox release)
```

File: docs/source/release-notes.html.md [hunk 4]
```diff
@@ -7,6 +7,25 @@ weight: 8
...
+
+Providers using all versions of the API can set the new `funded_place` field in the 'Accept an application' request body to true or false. They will also see the `funded_place` field in the following endpoint response bodies:
+
+'View all applications'
+
+'View a specific application'
+
+'View all participant data'
+
+'View a single participant's data'
```
Comment (cwrw) [inline docs/source/release-notes.html.md:24]:
```suggestion
- 'View all applications'

- 'View a specific application'

- 'View all participant data'

- 'View a single participant's data'
```

**Review by cwrw — COMMENTED** [review]
@padv2010 thank you for addressing comments. I think v3 is still missing a few places, let's chat tomorrow please. Thanks

File: swagger/v3/api_spec.json (hunk 2 above)
Comment (cwrw) [reply swagger/v3/api_spec.json:164]: can we please avoid this large change, can we jump on a call?

File: swagger/v3/component_schemas/NPQApplicationAcceptAttributesRequest.yml [hunk 5]
```diff
@@ -1,22 +1,7 @@
 description: "The NPQ Application acceptance request"
 type: object
-required:
```
Comment (cwrw) [inline swagger/v3/component_schemas/NPQApplicationAcceptAttributesRequest.yml:3]: Can we keep those please? as those are required for the acceptance on v3? and add the new funded place

File: swagger/v3/component_schemas/NPQApplicationAcceptRequest.yml [hunk 6]
```diff
@@ -7,4 +7,4 @@ example:
   data:
     type: "npq-application-accept"
     attributes:
-      schedule_identifier: "npq-leadership-spring"
```
Comment (cwrw) [inline swagger/v3/component_schemas/NPQApplicationAcceptRequest.yml:10]: same here

**Review by cwrw — APPROVED** [review]
thank you @padv2010

**Review by leandroalemao — APPROVED** [review]
tks @padv2010 🔝

**Pull request merged**